common/discovery-config-manager.cpp

```diff
@@ -0,0 +1,194 @@
+#include "discovery-config-manager.h"
+
+namespace fty::disco {
+
+ConfigDiscoveryManager& ConfigDiscoveryManager::instance()
+{
+    static ConfigDiscoveryManager inst;
+    return inst;
+}
+
+fty::Expected<void> ConfigDiscoveryManager::save(const std::string& path)
+{
+    if (!m_config.has_value()) {
+        return fty::unexpected("discovery config was not loaded => nothing to save");
+    }
+    m_config.value().save(path);
+
+    return {};
+}
+
+fty::Expected<ConfigDiscovery> ConfigDiscoveryManager::load(const std::string& path)
+{
+    auto tmp = ConfigDiscovery();
```

Comment:
Process description:
Create object;
Create second object;
Assign (possible as copy) to first one...
Of course, will be optimized, but better ti write:
```ConfigDiscovery tmp;``` looks better and no questions.